source/benchmarking/odm-benchmarking.md

```diff
@@ -0,0 +1,393 @@
+    off results
+- measuring 100,000 insertions one time -- in this case, the longer the timer runs, the higher the likelihood that an
+    external event occurs that affects the time of the run
+
+Therefore, we choose a middle ground:
+
+- measuring the same 10,000 insertions over 10 iterations -- each timing run includes enough operations that insertion
+    code dominates timing code; unusual system events are likely to affect only a fraction of the 10 timing measurements
+
+With 10 timings of inserting the same 10,000 models, we build up a statistical distribution of the operation timing,
+allowing a more robust estimate of performance than a single measurement. (In practice, the number of iterations could
+exceed 10, but 10 is a reasonable minimum goal.)
+
+Because a timing distribution is bounded by zero on one side, taking the mean would allow large positive outlier
+measurements to skew the result substantially. Therefore, for the benchmark score, we use the median timing measurement,
+which is robust in the face of outliers.
+
+Each benchmark is structured into discrete setup/execute/teardown phases. Phases are as follows, with specific details
+given in a subsequent section:
+
+- setup -- (ONCE PER TASK) something to do once before any benchmarking, e.g. construct a model object, load test data,
+    insert data into a collection, etc.
+- before operation -- (ONCE PER ITERATION) something to do before every task iteration, e.g. drop a collection, or
+    reload test data (if the test run modifies it), etc.
+- do operation -- (ONCE PER ITERATION) smallest amount of code necessary to execute the task; e.g. insert 10,000 models
+    one by one into the database, or retrieve 10,000 models of test data from the database, etc.
+- after operation -- (ONCE PER ITERATION) something to do after every task iteration (if necessary)
+- teardown -- (ONCE PER TASK) something done once after all benchmarking is complete (if necessary); e.g. drop the test
+    database
+
+The wall-clock execution time of each "do operation" phase will be recorded. We use wall clock time to model user
+experience and as a lowest-common denominator across ODMs. Iteration timing should be done with a high-resolution
+monotonic timer (or best language approximation).
+
+Unless otherwise specified, the number of iterations to measure per task is variable:
+
+- iterations should loop for at least 30 seconds cumulative execution time
+- once this 30 second minimum execution time is reached, iterations should stop after at least 10 iterations or 1 minute
+    cumulative execution time, whichever is shorter
+
+This balances measurement stability with a timing cap to ensure all tasks can complete in a reasonable time.
+
+For each task, the 10th, 25th, 50th, 75th, 90th, 95th, 98th and 99th percentiles will be recorded using the following
+algorithm:
+
+- Given a 0-indexed array A of N iteration wall clock times
```

Comment:
Some formatting for "A" and "N" would be helpful here for readability. You can simply use monospace formatting or something fancier like we have in the CSFLE spec: https://github.com/mongodb/specifications/blob/master/source/client-side-encryption/client-side-encryption.md#create-encrypted-collection-helper

Likewise for other special symbols below, such as "i" and "p".